ClimsoftVer4/ClimsoftVer4/frmNewFormDaily2.vb

```diff
@@ -100,4 +100,17 @@ Public Class frmNewFormDaily2
         End If
         clsDataConnection.SaveUpdate()
     End Sub
+
+    Private Sub btnDelete_Click(sender As Object, e As EventArgs) Handles btnDelete.Click
+        Dim cust As New form_daily2
+        cust.yyyy = "2018"
+        cust.stationId = "67755030"
+        cust.elementId = "2"
+        cust.mm = "2"
+        cust.hh = "1"
+        clsDataConnection.db.form_daily2.Attach(cust)
+        clsDataConnection.db.form_daily2.Remove(cust)
+        clsDataConnection.db.SaveChanges()
+        MsgBox("deleted")
+    End Sub
```

Comment:
This looks good. An alternative way and probably not so good for now could be to have a parameterized raw query or use a stored procedure to achieve a similar functionality. e.g

using a stored procedure named DeleteFormDailyByStationId
context.Blogs.SqlQuery("db.DeleteFormDailyByStationId @stId", stationId )

or
using raw Query(this could be parameterised to prevent against sql injection)
context.Database.ExecuteSqlCommand( 
        "DELETE FROM db.form_daily2 WHERE  yyyy="2018" AND stationId = "67755030" AND elementId = "2" AND mm = "2" AND hh = "1")  

But due to the fact that the team settled on somehow not using SQL this looks good.

Reply:
Well if we had to do your first suggestion then we may not know what rows to delete because we could have same station ID with different  element ID, hour day etc so we need then to pass in a complete filter. 
Using raw querry, looks sensible but why would we need to do that when  entity framework can do that for us easily? 
I wonder if we can create a general method that we can be passing a filter to then executes it? @volloholic  @dannyparsons  @shadrackkibet  what do you think ?

Reply:
@maxwellfundi in the first suggestion I meant to show how we could utilise procedures. Obviously the parameters of the procedure would include all the fields that make the composite key.